central/cloudsources/datastore/datastore_impl.go

```diff
@@ -97,7 +97,8 @@ func validateCloudSource(cloudSource *storage.CloudSource) error {
 	if cloudSource.GetName() == "" {
 		errorList.AddString("cloud source name must be defined")
 	}
-	if cloudSource.GetCredentials().GetSecret() == "" {
+	if cloudSource.GetCredentials().GetSecret() == "" &&
```

Comment:
Should we have a todo to remove it
Also should we show log with information about deprecated feature used?